src/ape_ethereum/ecosystem.py

```diff
@@ -1200,11 +1204,13 @@ def _enrich_contract_id(self, address: AddressType, **kwargs) -> str:
 
         kwargs["contract_type"] = contract_type
         if kwargs.get("use_symbol_for_tokens") and "symbol" in contract_type.view_methods:
-            # Use token symbol as name
-            contract = self.chain_manager.contracts.instance_at(
-                address, contract_type=contract_type
-            )
+            # Store contract_type in cache only if missing
+            if (contract := self.chain_manager.contracts.instance_at(address)) is None:
```

Comment:
I don't think `instance_at()` returns `None`, you probably want `.get_contract_type()` here.
```suggestion
            if (contract_type := self.chain_manager.contracts.get_contract_type(address)) is None:
```